guide/src/mocking_lros.md

```diff
@@ -0,0 +1,183 @@
+response.
+
+```rust,ignore
+{{#rustdoc_include ../samples/tests/mocking_lros.rs:auto-client-call}}
+```
+
+## Tests for manual polling with intermediate metadata
+
+Let's say our application code manually polls, and does some processing on
+partial updates.
+
+```rust,ignore
+{{#rustdoc_include ../samples/tests/mocking_lros.rs:manual-fn}}
+```
+
+We want to simulate how our application acts when it receives intermediate
+metadata. We can achieve this by returning in-progress operations from our mock.
+
+### Creating the `longrunning::model::Operation`
+
+The `BatchRecognize` RPC returns partial results in the form of a
+`speech::model::OperationMetadata`. Like before, we will need to pack this into
+the returned `longrunning::model::Operation`, but this time into the
+`Operation::metadata` field.
+
+```rust,ignore
+{{#rustdoc_include ../samples/tests/mocking_lros.rs:partial-op}}
+```
+
+### Test code
+
+First we define our mock class, which implements the
+[`speech::stub::Speech`][speech-stub] trait. Note that we override
+`get_operation()`. We will see why shortly.
+
+```rust,ignore
+{{#rustdoc_include ../samples/tests/mocking_lros.rs:manual-mockall-macro}}
+```
+
+Now in our test we create our mock, and set expectations on it.
+
+```rust,ignore
+{{#rustdoc_include ../samples/tests/mocking_lros.rs:manual-mock-expectations}}
+```
+
+These expectations will return partial results (25%, 50%, 75%), then return our
+desired final outcome.
+
+Now a few things you probably noticed.
+
+1. The first expectation is set on `batch_recognize()`, whereas all subsequent
+   expectations are set on `get_operation()`.
+
+   The initial `BatchRecognize` RPC starts the LRO on the server-side. The
+   server returns some identifier for the LRO. (This is the `name` field which
+   is omitted from the test code, for simplicity).
+
+   From then on, the client library just checks the status of that LRO. It does
+   this using the `GetOperation` RPC.
+
+   That is why we set expectations on different RPCs for the initial response
+   vs. all subsequent responses.
+
+1. Expectations are set in a [sequence].
+
+   This is a particularity of `mockall`. It allows `mockall` to distinguish
+   between different expectations.
+
+Finally, we create a client from the mock, call our function, and verify the
+response.
+
+```rust,ignore
+{{#rustdoc_include ../samples/tests/mocking_lros.rs:manual-client-call}}
+```
+
+## Simulating errors
+
+As before, simulating errors is no different than simulating a successful
+response.
+
+```rust,ignore
+{{#rustdoc_include ../samples/tests/mocking_lros.rs:error}}
+```
+
+Errors can be returned from the original RPC (`batch_recognize`) or from
+subsequent `get_operation` calls.
+
+______________________________________________________________________
+
+## Testing LROs - Full test suite
+
+Putting all this code together into a full test suite looks as follows:
+
+```rust,ignore,noplayground
+{{#include ../samples/tests/mocking_lros.rs:all}}
```

Comment:
This is long enough that maybe `rustdoc_include` works better ?

Reply:
Ahhh thanks. This helped resolved my confusion here: https://github.com/googleapis/google-cloud-rust/pull/1683#discussion_r2027126375

As a result, I can split this into three, self-contained sections. They are quite long, but I love that they can be copy-pasted as-is.